### DISCUSSION TITLE
Field value in Search results, recognized as a URL link.

### DISCUSSION THREAD

**Discussion by bitmapbulgaria** [hajkmap/Hajk#1185] [Q&A]
Hi, guys! I have made field in my PostgreSQL (string type), that have valid URL links inside. QGIS recognize the values as valid URL, but PgAdmin and Geoserver - no. This is just a regular string field.

I saw, that on the https://karta.halmstad.se/ there is active URL link in search results:
![image](https://user-images.githubusercontent.com/57766493/191918892-89b6b91f-3c64-40d1-b922-6c976c7a3fe8.png)

Is this a custom implementation of a functionality, or there is some settings in HAJK to point to a column values as URL link? I didn't find anything on the Admin Panel, or in the code for this.... It seem logical to have some event listener in HAJK, when you point on a value, but I can't find it. 
I discovered, that there is an Infobox URL setting for a layer, but my question is about per feature basis URL, stored in a field. Because, sometimes, if you have object, who have a document, attached to the object (for example- some regulation or some PDF generated for this object) it will be good idea to recognize the values in some field as "link-to-attached doc", or just to have setting in Admin Panel, that setup a field as container for URLs

**Comment by bitmapbulgaria** [comment]
After some digging the code, I found, that the logic for listening this is already applied into Hajk via plausible-tracker module and the job is done in /new-client/src/components/app.js, at rows 397 - 409:

 
`   * @summary Initiates the wanted analytics model (if any).
   * @description If Hajk is configured to track map usage, this method will
   * initialize the model and subscribe to two events ("analytics.trackPageView"
   * and "analytics.trackEvent").
   *
   * @memberof App
   */
  initiateAnalyticsModel() {
    this.analytics = new Analytics(
      this.props.config.mapConfig.analytics,
      this.globalObserver
    );
  }`
  
  So, the question is it bug or there is some plugin, should be activated?

**Reply by jacobwod** [reply]
Nah, I don't see what this would have to do with the Plausible tracker. 😄

**Comment by jool-kba** [comment]
Hi!

To extract values from the database, use curly brackets. For active links make sure the html a-tag is used.

Examples in infobox:
```
{column} 

or

<a href="{column}">Link</a>
```

**Comment by bitmapbulgaria** [comment]
Thanks, but for me it doesn't work, because, if I put {link} in Info click , Hajk omit other fields and values, an I got only the link in the search results.:
So, I tried {URL} (my column name is _URL_),
`      "layersInfo": [
        {
          "id": "ruse:addresses",
          "caption": "",
          "legend": "",
          "legendIcon": "",
          "infobox": "{URL}",
          "style": "",
          "queryable": true,
          "infoclickIcon": ""
        }
      ],
`
there and I have got:
![image](https://user-images.githubusercontent.com/57766493/193544521-d3fdbc95-2990-44a2-a64b-4f5805cca36d.png)
BUT, I need all needed info here:
(_"infobox": ""_)

![image](https://user-images.githubusercontent.com/57766493/193545017-339a6399-3f4a-44da-9071-0425c5974130.png)
As you can see, in the classic feature info result, the URL field is just a string... :( 
@jacobwod , can you guess, what I'm missing here?

**Reply by jacobwod** [reply]
Your screenshot shows what we call a "default attribute table": this mode is activated if none infobox definition could be found for the layer you clicked. (The idea is that if admin made a layer clickable but forgot to specify _what_ should be visible, we just display the current record from the database, as-is, in a table form.)

So it seems your infobox definition is ignored for some reason.

You can see how we setup our layers by doing some debugging on our site (the config can be found here: https://karta.halmstad.se/api/v2/config/ext). 

The part you refer to, in the initial post's screenshot, is this:
![Skärmavbild 2022-10-03 kl  13 31 04](https://user-images.githubusercontent.com/110222/193566881-a8c83d5c-9ea4-4b67-9d41-47fbb2819158.png)

Specifically, the infobox is styled using MarkDown and the part that creates the link is a normal MarkDown link, plus the value from database inside the curly brackets:
```markdown
"infobox": "{{if {byggnadurl}}}\n#### Länk\n[Byggnadsinformation från RAÄ]({byggnadurl}){{/if}}
```

(In addition, we have a header _plus_ an optional `if`-condition going on here, which ensure that we don't display the link if value of `byggnadsurl` is lacking. But you can just ignore the condition which gives you this basic line:
```markdown
"infobox": "[Byggnadsinformation från RAÄ]({byggnadurl})"
```

When configuring the infobox definition, I suggest you start with simple statements and make more complex combination when as you see how they work. So something like:
```markdown
This is just a string.

[This will be a link.](https://karta.halmstad.se)

This is a string with some value from the database: {columnNameInDB}.

[This will be a link with dynamic URL from DB.]({urlColumnInDB})
```


I hope this helps!

**Comment by bitmapbulgaria** [comment]
Many thanks! It's working.
